backend/dataall/cdkproxy/stacks/dataset.py

```diff
@@ -346,7 +405,7 @@ def __init__(self, scope, id, target_uri: str = None, **kwargs):
                 type='AWS::LakeFormation::Resource',
                 properties={
                     'ResourceArn': f'arn:aws:s3:::{dataset.S3BucketName}',
-                    'RoleArn': f'arn:aws:iam::{env.AwsAccountId}:role/{self.pivot_role_name}',
+                    'RoleArn': dataset.IAMDatasetAdminRoleArn,
```

Comment:
I did a few tests registering with the dataset admin role and all worked fine... to be honest I am a little surprised now that that was the case since the requirements in the [docs](https://docs.aws.amazon.com/lake-formation/latest/dg/registration-role.html) for the user-defined register role are:
- S3 Permissions to S3 Location - YES
- KMS Permissions - YES
- Trust Relationship to both Glue and LF - NO (only with Glue for dataset admin role)

I see your point this may have unintentional implications and can be seen as out of scope for this PR - will revert back to pivotRole for registering. At a later time we should look into removing PivotRole permissions for all datasets and instead use dataset roles to do the heavy lifting